lib/kitchen-sink/map.ex

```diff
@@ -138,4 +138,40 @@ defmodule KitchenSink.Map do
   def rename_key(_,_,_) do
     %{}
   end
+
+  def new_nested_map(key_list, value) do
+    key_list
+    |> Enum.reverse
+    |> Enum.reduce(value, fn (key, acc) -> %{key => acc} end)
+    |> Map.new
+  end
+
+  defp renamed_key(map) do
+    fn
+      ({old_key, [root_key | key_list]}) ->
+        value = Map.get(map, old_key)
+        {root_key, new_nested_map(key_list, value)}
+
+      ({old_key, new_key}) ->
+        value = Map.get(map, old_key)
+        {new_key, value}
+    end
+  end
+
+  @doc """
+  renames many keys in a map
+  """
+  def rename_keys(map, key_map, :prune) do
+    key_map
+    |> Enum.map(renamed_key(map))
+    |> Map.new
+  end
+  def rename_keys(map, key_map) do
```

Comment:
do not group functions that are not of same arity. You probably meant to pass prune as an option, then make it so as pass it as an option with default = false?